### PR TITLE
Fixed bad signing of ES requests issue #12

### PR DESCRIPTION
**Pull request by MelnykDmytroIF**
Fixed issue #12

### CODE REVIEW

**Comment by bastuijnman** [comment]
Hi @jeskew could you have a look at this pull request? We use this library pretty extensively within our organisation and it would be nice if we could fix our issue without forking 😃

Thanks in advance,

Bas Tuijnman
24i

File: src/ElasticsearchPhpHandler.php [hunk 1]
```diff
@@ -62,6 +59,8 @@ private function createPsr7Request(array $ringPhpRequest)
         $ringPhpRequest['headers'][$hostKey][0]
             = parse_url($ringPhpRequest['headers'][$hostKey][0])['host'];
 
+        $ringPhpRequest['headers']['Content-Type'] = 'application/json';
```
Comment (jeskew) [inline src/ElasticsearchPhpHandler.php:62]: Shouldn't this header have been set before the request was handed off to the HTTP handler? 

Also, I believe only requests with bodies should have a content-type header.

Comment (MelnykDmytroIF) [reply src/ElasticsearchPhpHandler.php:62]: @jeskew Without *Content-Type: application/json* Elastic client throws next exception:
Elasticsearch\Common\Exceptions\BadRequest400Exception
Message: Content-Type header is missing

It should be, but unfortunately, ElasticSearch PHP client doesn't attach this header for some reason.

Comment (jeskew) [reply src/ElasticsearchPhpHandler.php:62]: That implies that the unsigned request would not have been accepted even if authentication were disabled, so I don't think it belongs in this library. 

Is there a later handler that is adding this header?

**Pull request closed without merging**